PMSCore/Permission.cs

```diff
@@ -0,0 +1,58 @@
+﻿namespace PMSCore
+{
+    /// <summary>
+    /// Represents a user's specific access permissions.
+    /// Allows multiple permissions to be assigned to a single user.
+    /// </summary>
+    [Flags]
+    public enum Permission
+    {
+        None = 0,                // No permissions
+        CanCreateProject = 1,    // Permission to create projects
+        CanChangeProjectStatus = 2, // Permission to update project status
+        CanCreateTask = 4,       // Permission to create tasks
+        CanChangeTaskStatus = 8, // Permission to update task status
+        CanCreateUser = 16,      // Permission to create users
+        CanDeleteUser = 32,      // Permission to delete users
+        CanGenerateReport = 64,  // Permission to generate reports
+    }
```

Comment:
This is all we need here, every permissions are set on concrete user construction